builtin/k8s/platform.go

```diff
@@ -97,6 +99,57 @@ func (p *Platform) DefaultReleaserFunc() interface{} {
 	}
 }
 
+// ConfigSet is called after a configuration has been decoded
+// we can use this to validate the config
+func (p *Platform) ConfigSet(config interface{}) error {
+	c, ok := config.(*Config)
+	if !ok {
+		// this should never happen
+		return status.Errorf(codes.FailedPrecondition, "invalid configuration, expected *k8s.Config, got %T", config)
+	}
+
+	if len(c.DeprecatedPorts) > 0 {
+		return status.Errorf(codes.InvalidArgument, "invalid kubernetes platform config - the 'ports' field has been deprecated and removed "+
+			"in favor of Pod.Container.Port. Refer to Port documentation here: https://www.waypointproject.io/plugins/kubernetes#port")
+	}
+
+	// Some fields can be specified on pod.Container and at the top level, for convenience and for
+	// historical reasons. Validate that both are not set at once.
+	if c.Pod != nil && c.Pod.Container != nil {
+		containerOverlayErrStr := "%s defined multiple times - in top level config and in Pod.Container"
+		container := c.Pod.Container
+		err := utils.Error(validation.ValidateStruct(c,
+			validation.Field(&c.Probe,
+				validation.Empty.When(container.Probe != nil).Error(fmt.Sprintf(containerOverlayErrStr, "Probe")),
+			),
+			validation.Field(&c.ProbePath,
+				validation.Empty.When(container.ProbePath != "").Error(fmt.Sprintf(containerOverlayErrStr, "ProbePath")),
+			),
+			validation.Field(&c.Resources,
+				validation.Empty.When(container.Resources != nil).Error(fmt.Sprintf(containerOverlayErrStr, "Resources")),
+			),
+			validation.Field(&c.CPU,
+				validation.Empty.When(container.CPU != nil).Error(fmt.Sprintf(containerOverlayErrStr, "CPU")),
+			),
+			validation.Field(&c.Memory,
+				validation.Empty.When(container.Memory != nil).Error(fmt.Sprintf(containerOverlayErrStr, "Memory")),
+			),
+			validation.Field(&c.StaticEnvVars,
+				validation.Empty.When(container.StaticEnvVars != nil).Error(fmt.Sprintf(containerOverlayErrStr, "StaticEnvVars")),
+			),
+			validation.Field(&c.ServicePort,
+				validation.Empty.When(len(container.Ports) > 0).Error("Cannot define both 'service_port' and container 'port'. Use"+
+					" container 'port' multiple times for configuring multiple container ports"),
+			),
+		))
+		if err != nil {
+			return status.Errorf(codes.InvalidArgument, fmt.Sprintf("Invalid kubernetes platform plugin config: %s", err.Error()))
```

Comment:
Small nit: since it's already `Errorf` you can just do this, I think 🤔 
```suggestion
			return status.Errorf(codes.InvalidArgument, "Invalid kubernetes platform plugin config: %s", err.Error())
```